btb/hyper_parameter.py

```diff
@@ -224,35 +223,45 @@ def invert(inv_map, x):
             # Get a random category from the ones that had the given score
             return random.choice(np.vectorize(inv_map.get)(max_key))
 
-        inv_trans = np.vectorize(invert)(inv_map, x)
+        try:
```

Comment:
In this case, I followed the recommendations from [here](https://stackoverflow.com/questions/1952464/in-python-how-do-i-determine-if-an-object-is-iterable), where duck typing seems to be the most suggested option.

But you are right that invert function might raise a TypeError itself, even though it is not something likely to happen. Also, if it happened, the later invert(x) statement would equally raise it, even though it might be for another reason.

So, in any case, I'm OK with replacing it with `isinstance(e, collections.Iterable)`, so I'll do that if that's your preference.

Reply:
For sure, I understand and like that recommendation as well. But simply because the iteration is so deep within the try block, it seems dangerous here. However you choose to rewrite it is fine.